### PR TITLE
250 Implement findCityCouncilDistricts endpoint

### LINKED ISSUE #250: Implement findCityCouncilDistricts endpoint
## Acceptance criteria
- [x] Implement the `findCityCouncilDistricts` endpoint in compliance with #241
   - should be part of a `city-council-district` domain folder
- [x] Write end to end tests
- [x] Write unit tests

## helpful code: 
[the land use domain](https://github.com/NYCPlanning/ae-zoning-api/tree/main/src/land-use) 

## Related tickets
- blocked by #241

### PR DESCRIPTION
**Pull request by horatiorosa**
#### Description
implementation of city council district (ccd) endpoint
 - ccd controller, module, repository, service
 - cd unit and e2e tests
 - indentation fixes on openapi yaml
 - tweaks to README
 - regen ts and zod schemas

#### Tickets
 - Closes #250

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +90,9 @@ Finally, to run the api locally:
 npm run dev
 ```
 
+- View the documentation locally through port 3000
```
Comment (pratishta) [inline README.md:93]: Maybe we can say localhost:3000 to keep it consistent with the previous localhost port

File: src/city-council-district/city-council-district.repository.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @Inject(DB)
+    private readonly db: DbType,
+  ) {}
+
+  async findMany(): Promise<FindManyRepo> {
+    try {
+      return await this.db.query.cityCouncilDistrict.findMany({
+        columns: {
+          id: true,
+          liFt: false,
```
Comment (pratishta) [inline src/city-council-district/city-council-district.repository.ts:17]: Outta curiosity, why is it that we need to specify false for `liFt` but not for `mercatorFill` or `mercatorLabel` just to get the `id`?

File: openapi/openapi.yaml [hunk 3]
```diff
@@ -111,6 +111,30 @@ paths:
           $ref: "#/components/responses/NotFound"
         '500':
           $ref: "#/components/responses/InternalServerError"
+  /city-council-districts:
```
Comment (pratishta) [inline openapi/openapi.yaml:114]: This should not be part of the implementation PR. But once you pull changes from https://github.com/NYCPlanning/ae-zoning-api/pull/295, you should remove the 🚧 as part of this PR

**Review by pratishta — CHANGES_REQUESTED** [review]

File: src/city-council-district/city-council-district.repository.ts (hunk 2 above)
Comment (horatiorosa) [reply src/city-council-district/city-council-district.repository.ts:17]: It seems that without the `liFt` column, neither the `mercatorFill` or `mercatorLabel` are returned in the response. I am not sure why TBH. We can ask @TangoYankee for an explanation next week.

File: openapi/openapi.yaml (hunk 3 above)
Comment (horatiorosa) [reply openapi/openapi.yaml]: yeah, this was only bc I was blocked without having the documentation so I branched off from #295 to create this one.

File: src/city-council-district/city-council-district.repository.ts (hunk 2 above)
Comment (TangoYankee) [reply src/city-council-district/city-council-district.repository.ts:17]: The 'lift: false' is superfluous. specifying only 'id: true' will achieve the same result

Comment (horatiorosa) [reply src/city-council-district/city-council-district.repository.ts:17]: oh, i'll get rid of it (`liFt`)

Comment (horatiorosa) [reply src/city-council-district/city-council-district.repository.ts:17]: @pratishta **TIL**: it seems that when you specify what columns you want, the drizzle [docs](https://orm.drizzle.team/docs/rqb#partial-fields-select) state that ONLY those columns are returned. Not only that but when both true and false select options are present, all false options are ignored. So the `liFt: false` is bering ignored.

File: README.md [hunk 4]
```diff
@@ -90,6 +90,9 @@ Finally, to run the api locally:
 npm run dev
 ```
 
+- View the documentation locally via `localhost:3000`
```
Comment (TangoYankee) [inline README.md:93]: Could we reorder this a little? It's clearer to me to first make a point that the application is running on port `3000`. After that first point, we can make these sub points. sub-point 1, the root route `/` has the documentation built from the point where the application was run. sub-point 2, the `/api/{endpoint}` routes have the actual implementations

File: compose.yaml [hunk 5]
```diff
@@ -10,5 +10,9 @@ services:
       - POSTGRES_DB=${DATABASE_NAME}
     ports:
       - "8010:5432"
+    networks:
```
Comment (TangoYankee) [inline compose.yaml:13]: These changes to compose should be their own PR- this docker network technique has technically been approved for the ETL.

**Review by TangoYankee — CHANGES_REQUESTED** [review]
The implementation looks good. Next step would be to remove the construction emoji from the documentation because this PR implements it.

**Review by TangoYankee — APPROVED** [review]

**Review by pratishta — APPROVED** [review]
